README-local-install.md

```diff
@@ -0,0 +1,301 @@
+
+## Running tests
+
+- To run specific tests locally:
+    
+  `bin/test -k test_name`
+    
+- To run with a debugger:
+    
+  `bin/test --pdb`
+
+- Specific tests to run locally for schema changes:
+
+  `bin/test -k test_load_workbook`  
+  `bin/test -k test_load_schema`
+
+- Run the Pyramid tests with:
+
+  `bin/test -m "not bdd"`
+
+- Run the Browser tests with:
+
+  `bin/test -m bdd -v --splinter-webdriver chrome`
+
+- Run the Javascript tests with:
+
+  `npm test`
+
+- Or if you need to supply command line arguments:
+
+  `./node_modules/.bin/jest`
+
+- **Test ALL the things!**
+  
+  `bin/test -v -v --splinter-webdriver chrome && npm test`
+
+
+## Building Javascript and CSS
+
+
+Our Javascript is written using ES6 and JSX, so needs to be compiled using babel and webpack. Our CSS is written in the SCSS variant of [Sass](http://sass-lang.com/) and also needs compilation using webpack.
+
+- To re-build **production-ready** bundles, do:
+
+  `npm run build`
+
+  (This is also done as part of running buildout.)
+
+- To build **development** bundles and continue updating them as you edit source files, run:
+
+  `npm run dev`
+
+The development bundles are not minified, to speed up building. The above command runs continually in your terminal window and watches for changes in Javascript and SCSS files, rebuilding the bundles as you make changes.
+
+## Creating a demo machine
+
+
+- After buildout you (if you have the correct permissions) can run for a single-node "cluster":
+
+  `bin/deploy`
+  
+  
+- The script above will spin up a server in the AWS cloud with the current branch, and with a computed nameserver alias based on the branch and your username.  There are options to use a different branch and/or different instance name and also if you want to use AWS spot instances...and you can specify which AWS profile you want to use.   
+
+  
+- Deploy script help (how to specify name, instance size, etc):
+
+  `bin/deploy --help`
+      
+For all options, including setting up ES clusters (needed for full production).  After indexing (currently 8+hrs) the machine can be downsized at AWS to an m4.2xlarge, unless you are planning to submit significant data to it.
+
+- If you don't buildout locally, you can install denpendency boto3 first then deploy:
+
+```bash
+pip3 install boto3
+python src/encoded/commands/deploy.py --profile-name regulome
+```
+
+- The demo link uses this format: https://[instance_name].demo.regulomedb.org/regulome-search/
+
+## Linting your code within your code editor
+
+To set up linting with [Sublime Text 3](https://www.sublimetext.com/3) or [Visual Studio Code](https://code.visualstudio.com/), first install the linters:
+
+```bash
+pip3 install flake8
+npm install -g eslint
+npm install -g eslint-plugin-react
+```
+>:warning: _Note_: You don't have to use Sublime Text 3 but you *must* insure that linting in your editor behaves as it does in Sublime Text 3.
+
+
+
+### Sublime Text 3
```

Comment:
I commented about this in another place as well, but I'm not sure if we need these specific Sublime/VS comments. Though, they probably don't hurt.

Reply:
I personally use vs. I think we can just keep it until we have a better idea on style and formatting.